p2p/simulations/http_test.go

```diff
@@ -517,3 +625,107 @@ func TestHTTPSnapshot(t *testing.T) {
 		x.connEvent(nodes[0].ID, nodes[1].ID, true),
 	)
 }
+
+func TestMsgFilterPassMultiple(t *testing.T) {
+	// start the server
+	network, s := testHTTPServer(t)
+	defer s.Close()
+
+	client := NewClient(s.URL)
+	events := make(chan *Event, 100)
+	filterstr := "prb:0;test:0"
+	sub, err := client.SubscribeNetworkWithMsgFilter(events, false, filterstr)
+	if err != nil {
+		t.Fatalf("error subscribing to network events: %s", err)
+	}
+	defer sub.Unsubscribe()
+	// check we got all the events
+
+	nodeCount := 30
+	startNetwork(network, client, t, nodeCount)
```

Comment:
In order to get enough messages being sent, so that also the expected codes are received, this has been the easiest and fastest way I could see in order to make this happen. There may be a better alternative indeed, but this has worked for me.